akka-persistence/src/main/scala-2.12/akka/persistence/AbstractPersistentActor.scala

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.persistence
+
+import java.lang.{ Iterable ⇒ JIterable }
+
+import akka.actor.AbstractActor
+import akka.japi.{ Procedure, Util }
+import akka.persistence.{ Eventsourced, SnapshotOffer }
+
+/**
+ * Java API: an persistent actor - can be used to implement command or event sourcing.
+ *
+ * Scala 2.12 API version
+ */
+abstract class AbstractPersistentActor extends AbstractActor with PersistentActor with Eventsourced {
```

Comment:
This is not what we discussed and what you described.
For 2.12 we must:
* move the scala api `persist` methods to `PersistentActor` from `EventSourced`
* `AbstractPersistentActor extends AbstractActor with Eventsourced`

This probably means that we must have duplicate sources for AbstractPersistentActor, PersistentActor, Eventsourced.

We could get away with less duplication if we break binary compatibility for 2.11 also. Then we would not any version specific sourced at all, right? Tempting?

Do we have anything else extending `PersistentActor`? `AtLeastOnceDelivery`?